web/source/kmwosk.ts

```diff
@@ -32,6 +32,23 @@ namespace com.keyman {
 
     abstract getId(): string;
 
+    /**
+     * Uses canvas.measureText to compute and return the width of the given text of given font in pixels.
+     * 
+     * @param {String} text The text to be rendered.
+     * @param {String} font The css font descriptor that text is to be rendered with (e.g. "bold 14px verdana").
+     * 
+     * @see https://stackoverflow.com/questions/118241/calculate-text-width-with-javascript/21015393#21015393
+     */
+    static getTextWidth(text: string, font: string) {
+      // re-use canvas object for better performance
+      var canvas = OSKKey.getTextWidth['canvas'] || (OSKKey.getTextWidth['canvas'] = document.createElement("canvas"));
+      var context = canvas.getContext("2d");
```

Comment:
Is it possible and/or desirable to cache context as well?

Reply:
The code example I drew from didn't seem to think so.